session/src/main/scala/akka/stream/alpakka/cassandra/scaladsl/CassandraFlow.scala

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.cassandra.scaladsl
+
+import akka.NotUsed
+import akka.dispatch.ExecutionContexts
+import akka.stream.alpakka.cassandra.CassandraWriteSettings
+import akka.stream.scaladsl.{ Flow, FlowWithContext }
+import com.datastax.oss.driver.api.core.cql.{ BatchStatement, BatchType, BoundStatement, PreparedStatement }
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{ ExecutionContext, Future }
+
+/**
+ * Scala API to create Cassandra flows.
+ */
+object CassandraFlow {
+
+  /**
+   * A flow writing to Cassandra for every stream element.
+   * The element is emitted unchanged.
+   *
+   * @param session Cassandra session from `CassandraSessionRegistry`
+   * @param writeSettings settings to configure the write operation
+   * @param cqlStatement raw CQL statement
+   * @param statementBinder function to bind data from the stream element to the prepared statement
+   * @tparam T stream element type
+   */
+  def create[T](
+      session: CassandraSession,
+      writeSettings: CassandraWriteSettings,
+      cqlStatement: String,
+      statementBinder: (T, PreparedStatement) => BoundStatement): Flow[T, T, NotUsed] = {
+    Flow
+      .setup { (materializer, _) =>
+        implicit val executionContext: ExecutionContext = materializer.system.dispatcher
```

Comment:
can we use `session.ec` here instead? not that it's doing anything heavy with it but could be better performance to stay within the same ec